connect/runtime/src/test/java/org/apache/kafka/connect/storage/KafkaOffsetBackingStoreTest.java

```diff
@@ -460,6 +471,74 @@ public void testClientIds() {
+                capturedNewTopic.capture(), capturedAdminSupplier.capture());
+
+        store.configure(mockConfig(props));
+        store.start();
+
+        verify(storeLog).start();
+
+        doAnswer(invocation -> {
+            capturedConsumedCallback.getValue().onCompletion(null,
+                    new ConsumerRecord<>(TOPIC, 0, 0, 0L, TimestampType.CREATE_TIME, 0, 0,
+                            jsonConverter.fromConnectData("", null, Arrays.asList("connector1",
+                                    Collections.singletonMap("partitionKey", "partitionValue1"))), TP0_VALUE.array(),
+                            new RecordHeaders(), Optional.empty()));
+            capturedConsumedCallback.getValue().onCompletion(null,
+                    new ConsumerRecord<>(TOPIC, 1, 0, 0L, TimestampType.CREATE_TIME, 0, 0,
+                            jsonConverter.fromConnectData("", null, Arrays.asList("connector1",
+                                    Collections.singletonMap("partitionKey", "partitionValue1"))), TP1_VALUE.array(),
+                            new RecordHeaders(), Optional.empty()));
+            capturedConsumedCallback.getValue().onCompletion(null,
+                    new ConsumerRecord<>(TOPIC, 2, 0, 0L, TimestampType.CREATE_TIME, 0, 0,
+                            jsonConverter.fromConnectData("", null, Arrays.asList("connector1",
+                                    Collections.singletonMap("partitionKey", "partitionValue2"))), TP2_VALUE.array(),
+                            new RecordHeaders(), Optional.empty()));
+            capturedConsumedCallback.getValue().onCompletion(null,
+                    new ConsumerRecord<>(TOPIC, 3, 0, 0L, TimestampType.CREATE_TIME, 0, 0,
+                            jsonConverter.fromConnectData("", null, Arrays.asList("connector2",
+                                    Collections.singletonMap("partitionKey", "partitionValue"))), TP1_VALUE.array(),
+                            new RecordHeaders(), Optional.empty()));
+            storeLogCallbackArgumentCaptor.getValue().onCompletion(null, null);
+            return null;
+        }).when(storeLog).readToEnd(storeLogCallbackArgumentCaptor.capture());
+
+        // Trigger a read to the end of the log
+        store.get(Collections.emptyList()).get(10000, TimeUnit.MILLISECONDS);
+
+        Set<Map<String, Object>> connectorPartitions1 = store.connectorPartitions("connector1");
+        assertEquals(2, connectorPartitions1.size());
+
+        Set<Map<String, Object>> expectedConnectorPartition1 = new HashSet<>();
+        expectedConnectorPartition1.add(Collections.singletonMap("partitionKey", "partitionValue1"));
+        expectedConnectorPartition1.add(Collections.singletonMap("partitionKey", "partitionValue2"));
+
+        assertTrue(connectorPartitions1.containsAll(expectedConnectorPartition1));
+        assertTrue(expectedConnectorPartition1.containsAll(connectorPartitions1));
```

Comment:
Any reason to do this instead of `assertEquals(expectedConnectorPartition1, connectorPartitions1);`?